climada/hazard/base.py

```diff
@@ -760,6 +775,34 @@ def select(self, event_names=None, date=None, orig=None, reg_id=None, reset_freq
         haz.sanitize_event_ids()
         return haz
 
+    def select_tight_cent(self, buffer=0.0):
+        """
+        Reduce hazard to centroids to minimal box containing all non-zero
+        intensity points.
+
+        Parameters
+        ----------
+        buffer : float, optional
+            Buffer of box in degrees. The default is 0.0.
```

Comment:
The buffer is in those units in which the coordinates are, right?

Reply:
Also, since github removed the comments on the suggestion commit:

- Reducing a hazard to only those centroids that are non-zero is dangerous because then exposures might be assigned centroids with non-zero intensity, even though a centroid (now removed) with zero intensity would have been closer.

- The default buffer has been made the same as the default max distance for `assign_centroids`